R/R/zzz.R

```diff
@@ -1,22 +1,47 @@
-.onAttach <- function(libname, pkgname) {
-  tryCatch(
-    {
-      metaflow_attach()
-    },
-    error = function(e) {
-      packageStartupMessage("  * Call 'metaflow::install()' to finish installation * ")
-    }
+pkg.env <- new.env()
+
+pkg.env$configs <- list(
+  default = list(
+    metaflow_path = expression(reticulate::py_discover_config("metaflow")$required_module_path)
+  ),
+  batch = list(
+    metaflow_path = expression(path.expand(paste0(getwd(), "/metaflow")))
   )
-  invisible(NULL)
+)
+
+.onAttach <- function(libname, pkgname) {
+  if (py_module_available("metaflow")) {
+    metaflow_attach()
+  }
 }
 
 .onLoad <- function(libname, pkgname) {
-  tryCatch(
-    {
-      metaflow_load()
-    },
-    error = function(e) {
-    }
-  )
+  metaflow_python <- Sys.getenv("METAFLOW_PYTHON", unset = NA)
+  if (!is.na(metaflow_python))
+    Sys.setenv(RETICULATE_PYTHON = metaflow_python)
+  
+  if (!ensure_metaflow()) {
+    packageStartupMessage("* Metaflow Python dependencies not found *\n",
+                          "    - Call `install_metaflow()` to install into a new conda or virtualenv\n",
+                          "    - Set `METAFLOW_PYTHON` environment variable to use a custom metaflow installation.")
```

Comment:
Can you add more documentation around what should be the value of `METAFLOW_PYTHON`?